### PR TITLE
possible to disable auth integration entirely

### PR DESCRIPTION
**Pull request by peterbe**
Part of #1095

### CODE REVIEW

**Comment by peterbe** [comment]
I've already but this into my Content prototype:
```
"start": "cd node_modules/@mdn/yari && cross-env REACT_APP_DISABLE_AUTH=true yarn start"
```
:)

**Review by escattone — COMMENTED** [review]
I think this is a great idea. If we want to disable login, we'd set `REACT_APP_DISABLE_AUTH=true` prior to building correct? If so, I think we should add `REACT_APP_DISABLE_AUTH=true` somewhere around https://github.com/mdn/yari/blob/646697ade38d7da7efb7a1793f7f5b5523c515db/.github/workflows/pr-build.yml#L90 right?

File: docs/envvars.md [hunk 1]
```diff
@@ -195,3 +195,11 @@ the link to sign in with Kuma needs this to be set.
...
 `REACT_APP_KUMA_HOST=localhost.org:8000` and now sign in links go to that
 host name instead. That means you can log in _from_ Yari with a single click.
+
+### `REACT_APP_DISABLE_AUTH`
+
+**Default: `false`**
+
+This entirely removes the sign-in and then `whoami` XHR fetching.
+Useful when using Yari purely for doing content editing and not interested
+in testing how Yari works for end users.
```
Comment (schalkneethling) [inline docs/envvars.md:205]:
```suggestion
This removes sign-in and `whoami` XHR fetching.
Useful when using Yari purely for content editing as authentication is then not required.
```

File: client/src/header/login.tsx [hunk 2]
```diff
@@ -7,8 +7,20 @@ import { getAuthURL } from "./auth-link";
 import { useUserData } from "../user-context";
 
 import avatarImage from "./avatar.png";
+import { DISABLE_AUTH } from "../constants";
 
 export default function Login() {
+  // For example, if you're Yari purely for previewing your content edits,
```
Comment (schalkneethling) [inline client/src/header/login.tsx:13]:
```suggestion
  // For example, if you're using Yari purely for previewing your content edits,
```

**Review by schalkneethling — COMMENTED** [review]
Two tiny nits. Other than that lgtm and makes sense. Thanks, @peterbe!

**Review by escattone — APPROVED** [review]
Thanks @peterbe!

**Pull request merged**